hkpy/hkbase/hkbase.py

```diff
@@ -158,10 +160,40 @@ def get_repositories(self) -> List[HKRepository]:
         except Exception as err:
             raise HKpyError(message='Could not retrieve existing repositories.', error=err)
 
+
+    @staticmethod
+    def get_auth_token(
```

Comment:
Perhaps this would be better suited with the utils' module.

Reply:
Why does hkpy need to offer this functionality?

Reply:
I've placed it into HKBase, because in HKLib I've placed in HKDatasource.
This logic was being replicated in several places and for me it should either be within HKBase or the core libs.

Reply:
Fair enough. I still believe `utils.misc` might be a better fit.